### PR TITLE
Add github action for building documentation

### PR DESCRIPTION
**Pull request by weierophinney**
Basic idea for generalized documentation build action using GitHub Actions.

### CODE REVIEW

File: github-actions/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+As an example:
+
+```yaml
+- name: Docs
+  uses: laminas/github-actions/docs@master
+  with:
+    emptyCommits: false
+    publishDir: doc/html
+    siteUrl: https://api-tools.laminas.dev
```
[the remark this answers is not in the document]
Comment (weierophinney) [reply github-actions/docs/README.md]: I was just using it as an example. I'll change it so it's less confusing.

File: github-actions/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+## Example workflow
+
+```yaml
+name: docs
+
+on:
+  push:
+    branches:
+      - master
+  repository_dispatch:
```
[the remark this answers is not in the document]
Comment (weierophinney) [reply github-actions/docs/README.md]: It does! See https://github.com/laminas/laminas-db/commit/2edb45791fa89dbf323d71d287a7aab1bcb2575f - that was triggered via this call.

That event is only for people with write access to the repo as well, so it cannot get triggered by non-organization members.

File: github-actions/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+jobs:
+  build-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build and deploy documentation
+        uses: laminas/documentation-theme/github-actions/docs@master
```
[the remark this answers is not in the document]
Comment (weierophinney) [reply github-actions/docs/README.md]: Yes. Format is `{org}/{repo}/{path}@{ref}`.

File: github-actions/docs/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    fi
+fi
+
+PUBLISH_REPOSITORY=${GITHUB_REPOSITORY}
+PUBLISH_BRANCH=gh-pages
+
+print_info "Deploy to ${PUBLISH_REPOSITORY}@${PUBLISH_BRANCH} from directory ${INPUT_PUBLISH_DIR}"
+
+print_info "Cloning documentation theme"
+git clone git://github.com/laminas/documentation-theme.git ${GITHUB_WORKSPACE}/documentation-theme
```
[the remark this answers is not in the document]
Comment (weierophinney) [reply github-actions/docs/entrypoint.sh]: Not sure - I'm uncertain how we'd reference the code from this repo, to be honest - or if it gets the entire repo, or just the stuff from the action's subdirectory. This ensures it always works.

File: github-actions/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+with no errors.
+
+As an example:
+
+```yaml
+- name: Docs
+  uses: laminas/github-actions/docs@master
+  with:
+    emptyCommits: false
+    publishDir: doc/html
```
[the remark this answers is not in the document]
Comment (weierophinney) [reply github-actions/docs/README.md]: lamians-eventmanager, lamians-permissions-rbac, and lamians-soap appear to still have a `doc` directory instead...

File: github-actions/docs/entrypoint.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (geerteltink) [reply github-actions/docs/entrypoint.sh]: IIRC it only gets the action and it subdirectories.

File: github-actions/docs/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (michalbundyra) [reply github-actions/docs/README.md]: Oh, so maybe we did not... Thx 👍

**Pull request merged**